atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/KvsSweepQueuePersister.java

```diff
@@ -18,38 +18,22 @@
 
 import java.util.List;
 
-import com.palantir.atlasdb.keyvalue.api.KeyValueService;
-import com.palantir.atlasdb.schema.TargetedSweepSchema;
-import com.palantir.atlasdb.table.description.Schemas;
-
 public final class KvsSweepQueuePersister implements MultiTableSweepQueueWriter {
-    private final KeyValueService kvs;
-    private final SweepableCellsWriter sweepableCellsWriter;
-    private final SweepableTimestampsWriter sweepableTimestampsWriter;
-
-    private KvsSweepQueuePersister(KeyValueService kvs, SweepableCellsWriter cells, SweepableTimestampsWriter ts) {
-        this.kvs = kvs;
-        this.sweepableCellsWriter = cells;
-        this.sweepableTimestampsWriter = ts;
-    }
+    private final SweepableCells sweepableCells;
+    private final SweepableTimestamps sweepableTimestamps;
 
-    public static KvsSweepQueuePersister create(KeyValueService kvs) {
-        WriteInfoPartitioner partitioner =  new WriteInfoPartitioner(kvs);
-        SweepableCellsWriter cellsWriter = new SweepableCellsWriter(kvs, partitioner);
-        SweepableTimestampsWriter timestampsWriter = new SweepableTimestampsWriter(kvs, partitioner);
-        KvsSweepQueuePersister persister = new KvsSweepQueuePersister(kvs, cellsWriter, timestampsWriter);
-        persister.initialize();
-        return persister;
+    private KvsSweepQueuePersister(SweepableCells cells, SweepableTimestamps ts) {
+        this.sweepableCells = cells;
+        this.sweepableTimestamps = ts;
     }
 
-    // todo(gmaretic): remove this
-    public void initialize() {
-        Schemas.createTablesAndIndexes(TargetedSweepSchema.INSTANCE.getLatestSchema(), kvs);
+    public static KvsSweepQueuePersister create(KvsSweepQueueTables tables) {
+        return new KvsSweepQueuePersister(tables.sweepableCells, tables.sweepableTimestamps);
```

Comment:
Here, two internals of `tables` are exposed, just so that we can call the same method (`enqueue`) on them at the same time, with the same argument. `KvsSweepQueueTables` itself doesn't do anything.

Let's keep `tables` as a field here and call `tables.enqueue(writes)` below, which will then enqueue the writes for both `sweepableCells` and `sweepableTimestamps`. Then, `KvsSweepQueueTables` has more purpose, and we're closer to making its fields private.

Reply:
Will merge these two into one class